src/pages/RepoPage/FailedTestsTab/FailedTestsTab.spec.tsx

```diff
@@ -1,66 +1,127 @@
-  (initialEntries = '/gh/codecov/cool-repo/tests/new') =>
+  (initialEntries = '/gh/codecov/cool-repo/tests') =>
   ({ children }) =>
     (
-      <MemoryRouter initialEntries={[initialEntries]}>
-        <Route
-          path={[
-            '/:provider/:owner/:repo/tests/new',
-            '/:provider/:owner/:repo/tests/new/codecov-cli',
-            '/:provider/:owner/:repo/tests/random-path',
-          ]}
-        >
-          <Suspense fallback={null}>{children}</Suspense>
-        </Route>
-        <Route
-          path="*"
-          render={({ location }) => {
-            testLocation = location
-            return null
-          }}
-        />
-      </MemoryRouter>
+      <QueryClientProvider client={queryClient}>
+        <MemoryRouter initialEntries={[initialEntries]}>
+          <Route
+            path={[
+              '/:provider/:owner/:repo/tests',
+              '/:provider/:owner/:repo/tests/new',
+              '/:provider/:owner/:repo/tests/new/codecov-cli',
+              '/:provider/:owner/:repo/tests/:branch',
+            ]}
+            exact
+          >
+            <Suspense fallback={null}>{children}</Suspense>
+          </Route>
+          <Route
+            path="*"
+            render={({ location }) => {
+              testLocation = location
+              return null
+            }}
+          />
+        </MemoryRouter>
+      </QueryClientProvider>
     )
 
+beforeAll(() => {
+  server.listen()
+})
+
+afterEach(() => {
+  queryClient.clear()
+  server.resetHandlers()
+})
+
+afterAll(() => {
+  server.close()
+})
+
+const mockRepoOverview = ({ testAnalyticsEnabled = false }) => {
+  return {
+    owner: {
+      repository: {
+        __typename: 'Repository',
+        private: false,
+        defaultBranch: 'main',
+        oldestCommitAt: '2022-10-10T11:59:59',
+        coverageEnabled: true,
+        bundleAnalysisEnabled: true,
+        languages: ['javascript'],
+        testAnalyticsEnabled,
+      },
+    },
+  }
+}
+
 describe('FailedTestsTab', () => {
-  function setup() {
-    const user = userEvent.setup()
-    const hardRedirect = jest.fn()
-    mockedUseRedirect.mockImplementation((data) => ({
-      hardRedirect: () => hardRedirect(data),
-    }))
-
-    return { hardRedirect, user }
+  const hardRedirect = jest.fn()
+  mockedUseRedirect.mockImplementation((data) => ({
+    hardRedirect: () => hardRedirect(data),
+  }))
+
+  function setup({ testEnabled = false }: { testEnabled?: boolean }) {
+    server.use(
+      graphql.query('GetRepoOverview', (req, res, ctx) => {
+        if (testEnabled) {
+          return res(
+            ctx.status(200),
+            ctx.data(mockRepoOverview({ testAnalyticsEnabled: true }))
+          )
+        }
+        return res(ctx.status(200), ctx.data(mockRepoOverview({})))
+      })
+    )
+
+    return { user: userEvent.setup({}) }
   }
 
   it('renders intro', () => {
-    setup()
+    setup({})
```

Comment:
for the future, you can just default the setup param to `{}` and you won't have to update it everywhere else

Reply:
for sure! Will keep that in mind